inline/src/lexer/token.rs

```diff
@@ -294,14 +297,12 @@ impl Token {
                 any_other_kind => (any_other_kind, any_other_kind),
```

Comment:
maybe `unreachable!()` would be better here, since the span would not be correct with the same kind twice, and the `is_ambiguous()` check should prevent this case anyways